clients/drcachesim/tracer/tracer.cpp

```diff
@@ -1500,19 +1500,10 @@ event_delay_app_instruction(void *drcontext, void *tag, instrlist_t *bb, instr_t
                              opnd_create_reg(scratch1)));
     MINSERT(bb, instr,
             XINST_CREATE_jump_cond(drcontext, DR_PRED_LT, opnd_create_instr(skip_call)));
-#        endif
 
     /* hit_instr_count_threshold does not always return. Restore scratch registers and
      * aflags.
      */
-#        ifdef X86_64
```

Comment:
I'd expect a XXX/FIXME/TODO #4711 comment explaining that this needs to be done for x86 too, but this fix seems to cause instability and so we're leaving x86 as technically broken to keep our tests green until someone is able to figure out what is going on.

Reply:
I would put this type of explanation in the commit description too, and in the (left open) issue: we're leaving x86 with buggy code b/c the fix seems to cause problems we do not yet understand.

Reply:
Got it. I converted it back to a draft until I finished testing it.

Reply:
Done.